src/app/curriculum/curriculum.component.ts

```diff
@@ -1,12 +1,70 @@
 import { Component } from '@angular/core';
-import { PublicLibraryComponent } from '../modules/library/public-library/public-library.component';
+import { ConfigService } from '../services/config.service';
 import { LibraryFiltersComponent } from '../modules/library/library-filters/library-filters.component';
+import { LibraryService } from '../services/library.service';
 import { MatIconModule } from '@angular/material/icon';
+import { MatTabsModule } from '@angular/material/tabs';
+import { PersonalLibraryComponent } from '../modules/library/personal-library/personal-library.component';
+import { PublicLibraryComponent } from '../modules/library/public-library/public-library.component';
+import { UserService } from '../services/user.service';
 
 @Component({
-  selector: 'curriculum',
-  imports: [LibraryFiltersComponent, MatIconModule, PublicLibraryComponent],
-  templateUrl: './curriculum.component.html',
-  styleUrl: './curriculum.component.scss'
+  imports: [
+    LibraryFiltersComponent,
+    MatIconModule,
+    MatTabsModule,
+    PersonalLibraryComponent,
+    PublicLibraryComponent
+  ],
+  styleUrl: './curriculum.component.scss',
+  templateUrl: './curriculum.component.html'
 })
-export class CurriculumComponent {}
+export class CurriculumComponent {
+  protected showMyUnits: boolean;
+  private numMyUnitsVisible: number = 0;
+  private numPublicUnitsVisible: number = 0;
+
+  constructor(
+    protected configService: ConfigService,
+    private libraryService: LibraryService,
+    private userService: UserService
+  ) {}
+
+  ngOnInit(): void {
+    this.setShowMyUnits();
+    this.getLibraryProjects();
+    this.subscribeNumUnitsVisible();
+  }
+
+  private setShowMyUnits(): void {
+    this.userService.getUser()?.subscribe((user) => {
+      this.showMyUnits = user.roles && user.roles.includes('teacher');
+    });
```

Comment:
Is there a need to check the result of calling getUser() since the function always returns a BehaviorSubject? Also, use userService.isTeacher()?
```suggestion
    this.userService.getUser().subscribe(() => {
      this.showMyUnits = this.userService.isTeacher();
    });
```